### ISSUE TITLE
Exporting a function with pre-defined attributes

### ISSUE THREAD

**Issue by alm0st907** [Azure/azure-functions-openapi-extension#132]
I am looking to write and pre-package Azure Functions to be consumed within my organization, and we would like these to be visible via the Swagger endpoint.

I played around with the source to understand more about how the functions are exported into the consuming service (such as the sample projects) but was unable to get the attributes such as OpenApiOperation to transfer across and show in the consuming service's Swagger endpoint. Is this possible currently, or a future use case that will be supported?

As an example:

The RenderSwaggerDocument route is added via this package to the consuming service, but is not exposed in the resulting swagger page.

If the various attributes for OpenApiOperation, OpenApiRequestBody, etc were added to that route, I would like it to show up in the consuming service's swagger page. Currently if I do this, the route and its information is not rendered in the resulting swagger page.

**Comment by justinyoo** [comment]
@alm0st907 Thanks for the issue!

All 4 endpoints starting with `Render*` are only displayed on your local dev environment, and encapsulated when deployed to Azure, which is by design. But all other endpoints written by you are exposed both on your local dev and Azure.

Is that what you're looking for?

**Comment by alm0st907** [comment]
In our use case, we write a package which adds some diagnostic and health check endpoints, and any function project which adds the package adds those endpoints, just like this package. 

Originally, with the Aliencube package, we wrote our own UI render trigger, and we could add the additional documentation as we built that trigger. We migrated from Aliencube to this package to make use of the OAuth additions. We ran into a separate issue with our custom render endpoint and OAuth (captured here #137) 

We would like to be able to have those endpoints documented and exposed in the Swagger UI, just like any other additional endpoints we would write and annotate would be. In the longer term, we would like to pre-package and annotate triggers which can be pushed via nugets and ensure they are added to the Swagger page. We can't tap into the document building process for the render as far as we know for now with this version of the package. 


As of right now, we are in the position of either 1) having full Swagger UI documentation for all endpoints but broken OAuth (blocking) or 2) Have incomplete Swagger documentation but working OAuth (not ideal, but working)

**Comment by justinyoo** [comment]
If you want to include some pre-packaged endpoints, you might want to look at [this code](https://github.com/Azure/azure-functions-openapi-extension/blob/main/src/Microsoft.Azure.WebJobs.Extensions.OpenApi/OpenApiTriggerFunctionProvider.cs) and [this code](https://github.com/Azure/azure-functions-openapi-extension/blob/main/src/Microsoft.Azure.WebJobs.Extensions.OpenApi/OpenApiTriggerFunctions.cs) and [this code](https://github.com/Azure/azure-functions-openapi-extension/blob/main/src/Microsoft.Azure.WebJobs.Extensions.OpenApi/OpenApiWebJobsStartup.cs) as examples.

**Comment by alm0st907** [comment]
Yes, I have looked at that code and generally understand it. I have written some of my own functions to pre-package endpoints...the key issue is integrating them into the Swagger UI and the generated JSON from the endpoint. 

The issue is that those pre-packaged functions will not show up in the `api/swagger.json` route,  nor the Swagger UI page to interact with. 

When we wrote our own trigger endpoint with the Aliencube package, our endpoint looked something like this

```
            var result = await swaggerConfig.Document
                .InitialiseDocument()
                .AddMetadata(swaggerConfig.OpenApiInfo)
                .AddServer(req, swaggerConfig.HttpSettings.RoutePrefix)
                .AddNamingStrategy(swaggerConfig.NamingStrategy)
                .AddVisitors(swaggerConfig.GetVisitorCollection())
                .Build(swaggerConfig.Assembly)
                .AddEventHandlers(req, swaggerConfig.Assembly)
// begin custom code to add the document to the spec
                .AddHealthcheck()
                .AddPing()
// end custom code
                .RenderAsync(swaggerConfig.SpecVersion, format)
                .ConfigureAwait(false);
```


Now that this package owns the rendering endpoint, we do not have access to add to the document like above, which could add our pre-packaged code into the Swagger UI page. 


In some local digging and playing with the samples, attributes to the pre-packaged endpoints do not actually do anything.
For instance, the various `Render*` endpoints have `[OpenApiIgnore]` but with this removed, and cleaning and rebuilding solution and samples, those endpoints do not show up in the OpenAPI spec, and if I add attributes as well, it does not render in the Swagger UI. 


Will adding pre-packaged code into the Swagger UI going to be a supported use case?

**Comment by justinyoo** [comment]
If you want to add your pre-packaged endpoints, say the HealthCheck one, those endpoints also need to follow the same as the other endpoints - adding `[OpenApiOperation(...)]`, `[OpenApiSecurity(...)]`, `[OpenApiParameter(...)]`, etc. Have you applied them all to your pre-packaged endpoints?

**Comment by alm0st907** [comment]
Yes, I have applied those attributes, and they do not transfer across from the originating package into the consumer and are not in the resulting Swagger page.

**Comment by justinyoo** [comment]
Alright. That's because it only looks for the current runtime DLL and OpenAPI package, not your pre-package DLL. Let me find out how we can put additional DLLs like your pre-package ones.

**Comment by justinyoo** [comment]
Related to #351

**Comment by justinyoo** [comment]
The feature is now in place for the out-of-proc worker (dotnet-isolated). It will be released in the upcoming release v1.2.0. Stay tuned!
